src/utils/allocators.zig

```diff
@@ -2,6 +2,269 @@ const std = @import("std");
 const builtin = @import("builtin");
 const sig = @import("../sig.zig");
 
+/// very similar to RecycleFBA but with a few differences:
+/// - this uses an explicit T type and only returns slices of that type (instead of a generic u8)
+/// - additional memory blocks are supported (instead of using a fixed-buffer-allocator approach)
+/// - record serialization is supported for fast loading this memory back
+/// - the global index is also returned on allocation to help with additional book-keeping
+/// NOTE: T needs to have a .DEFAULT field that is used to zero out the memory allocated in `append()`
```

Comment:
I think with the api updated to take `default_init`, this comment about T needing to have a .DEFAULT is now outdated.